app/scripts/contentscript.js

```diff
@@ -331,19 +419,32 @@ const destroyLegacyExtensionStreams = () => {
 };
 
 /**
- * Resets the extension stream with new streams to channel with the in page streams,
- * and creates a new event listener to the reestablished extension port.
+ * When the extension background is loaded it sends the EXTENSION_MESSAGES.READY message to the browser tabs.
+ * This listener/callback receives the message to set up the streams after service worker in-activity.
+ *
+ * @param {object} msg
+ * @param {string} msg.name - custom property and name to identify the message received
+ * @returns {Promise|false}
  */
-const resetStreamAndListeners = () => {
-  extensionPort.onDisconnect.removeListener(resetStreamAndListeners);
+const onMessageSetUpExtensionStreams = (msg) => {
+  if (msg.name === EXTENSION_MESSAGES.READY) {
+    setupExtensionStreams();
+    setupLegacyExtensionStreams();
+    return Promise.resolve(`MetaMask: handled ${EXTENSION_MESSAGES.READY}`);
+  }
 
-  destroyExtensionStreams();
-  setupExtensionStreams();
+  return false;
+};
 
```

Comment:
Hey @digiwand : I know it is complex, but could you test phishing stream and legacy stream changes.

Reply:
Thanks @seaona for taking this head on! As discussed earlier, we will handle the support for the phishing stream logic in a separate PR (see https://github.com/MetaMask/metamask-extension/issues/15987)

@seaona has found that terminating the service worker breaks the phishing warning page logic